tests/codegen-llvm/pclmulqdq-target-feature-inlining.rs

```diff
@@ -0,0 +1,33 @@
+// Test for pclmulqdq intrinsics inlining across target_feature boundaries
+// Issue: https://github.com/rust-lang/rust/issues/139029
+//
+// When a function with target_feature calls another function without matching
+// target_feature attributes, the intrinsics may not inline properly, leading
+// to function calls instead of direct instructions.
+
+//@ only-x86_64
+//@ compile-flags: -C opt-level=3
+
+#![crate_type = "lib"]
+
+use std::arch::x86_64 as arch;
+
+// CHECK-label: @reduce128_caller
+// CHECK-NOT: fastcc
+// CHECK-NOT: define internal
```

Comment:
I'd suggest being more explicit here, something along these lines:
```suggestion
// CHECK-NOT: call
// CHECK: call <2 x i64> @llvm.x86.pclmulqdq
// CHECK: call <2 x i64> @llvm.x86.pclmulqdq
// CHECK-NOT: call
```